zkfold-ledger/src/ZkFold/Symbolic/Ledger/Types/Update.hs

```diff
@@ -2,30 +2,33 @@ module ZkFold.Symbolic.Ledger.Types.Update where
 
 import           Prelude                                hiding (Bool, Eq, length, splitAt, (*), (+))
 
-import           ZkFold.Symbolic.Ledger.Types.Basic
-import           ZkFold.Symbolic.Ledger.Types.Contract  (ContractId)
+import           ZkFold.Symbolic.Data.Bool              (Bool)
+import           ZkFold.Symbolic.Data.Combinators       (RegisterSize (Auto))
+import           ZkFold.Symbolic.Data.List              (List)
+import           ZkFold.Symbolic.Data.UInt              (UInt)
+import           ZkFold.Symbolic.Ledger.Types.Contract  (ContractData, ContractId)
 import           ZkFold.Symbolic.Ledger.Types.Hash      (Hash)
-import           ZkFold.Symbolic.Ledger.Types.Input     (Input)
 import           ZkFold.Symbolic.Ledger.Types.OutputRef (TransactionId)
 
 -- TODO: Add contract public data to the update.
 
 -- | Block hash of the corresponding block.
-type UpdateId context = Hash context
+type UpdateId = Hash
+
+type AddressIndex = UInt 40 Auto
 
--- | Update is a public data contained in a block.
--- TODO: This has been significantly redesigned. See the internal PDF docs.
--- Instead of "public inputs and outputs", we now have "online and offline transactions".
 data Update context = Update
```

Comment:
1. The Merkle tree root for online transactions is missing. Online transactions require two-round communication between the address owner and the aggregation server in which the address owner gets the path to their leaf in the Merkle tree.
3. The list of released indices is missing. The idea is that if the money from a particular address is completely spent, then the address is removed from the index. That way, we are protected against index spam attacks.

Reply:
1. Added `updateId` as a record field. I had misunderstood that it was derivable somehow from the other fields.

2. Added `updateIndicesReleased` as a record field.

Reply:
`updateId` is not a good name for it because it may be confused with the hash of the `Update` type. This field is a Merkle tree root of the online transactions in the update.

Reply:
changed the name to `updateOnlineTxsRoot`